change/@ni-nimble-components-f50c155f-c243-4397-937d-fc4fa99c2aec.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
I'd like to suggest a little bit of work to make it easier to leverage this infrastructure for other components:
1. Deploy the `/nimble-components/performance` directory in the Chromatic build so a user can visit it for each build (I believe that's configured in https://github.com/ni/nimble/blob/main/packages/site/.eleventy.js)
2. Link to it from [Storybook Getting Started](https://nimble.ni.dev/storybook/?path=/docs/getting-started--docs) alongside the Angular and Blazor apps
3. Change `performance/index.html` to be a simple table of contents that links to individual performance pages and move the existing contents to `performance/wafer-map.html`
4. Add some docs to `nimble-components/CONTRIBUTING.md` describing how to add a new performance test and look at its results
5. Add some docs to https://github.com/ni/nimble/blob/main/CONTRIBUTING.md#pull-request-policies mentioning what is enforced by the pipeline and what to do if it fails

I'm open to feedback from @rajsite and others if any of this seems like the wrong direction or from you if it seems too complicated.

Reply:
Milan mentioned he had some more ideas around this (e.g. separating the performance tests into a separate package) but we didn't get a chance to discuss them yet today. If it makes sense for you, I'd suggest waiting to address this feedback until we have a chance to agree on a concrete proposal. We'll aim to share something back before end of day tomorrow.

Reply:
I made a refactoring of most of the code, that’s why @munteannatan , @atmgrifter00, I reset your votes. The wafer performance tests are now aligned with the existing ones of the other components. The only concern may be that the wafer takes a few seconds to load, we may consider making it smaller. Some of the above comments may no longer be necessary, I’m waiting for a second review.

Reply:
I have step by step recommendations to make it easy for you to implement and to show you're following existing patterns, v it should be discussed before you go a different direction

> The wafer performance tests are now aligned with the existing ones of the other components.

The other components are not trying to run performance tests, they are using lighthouses other capabilities on page size measurement, accessibility scores, etc. It is a poor performance test because Angular adds overhead and all the other controls lacking at the same time add overhead.

Reply:
@rajsite @jattasNI I think I implemented most of your feedback, with 2 exceptions, Jesse's comments 5 and 3, 5: I discussed with @munteannatan and he wants, at least the wafer performance tests pipeline, to not fail anytime. 3: now index.html doesn't contain wafer-specific code, the wafers are appended from wafer-map.ts to index.html, would you want a different approach?

Reply:
I addressed 3 in the changes I pushed by creating an index page for the benchmarks that is linked from storybook. This avoids a storybook change every time a new benchmark is added.

Agree that no status checks from the performance tests should fail the pipeline for now. Currently it is just being used to track values.

Reply:
I think with the changes I've pushed everything in the thread is resolved.